### PR TITLE
2fa code boxes input

### PR DESCRIPTION
**Pull request by pvicensSpacedev**
### Summary of Changes

- Changed the single input in  onboarding 2fa to be a series of boxes

### Demo or Before/After Pics

https://github.com/gallery-so/gallery/assets/112896251/8f3cee78-c2fb-4d61-8fa9-4c5926d92ea0

### Checklist

Please make sure to review and check all of the following:

- [ ] I've tested the changes and all tests pass.
- [ ] (if web) I've tested the changes on various desktop screen sizes to ensure responsiveness.
- [ ] (if mobile) I've tested the changes on both light and dark modes.

### CODE REVIEW

**Review by kaitoo1 — APPROVED** [review]

File: apps/mobile/src/screens/Login/Onboarding2FAInput.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    setCode(newCode);
+
+    if (text && index < 5) {
+      inputs.current[index + 1]?.focus();
+    }
+
+    if (newCode.join('').length === 6) {
+      onCodeComplete(newCode.join(''));
+    }
+  };
```
Comment (Robinnnnn) [inline apps/mobile/src/screens/Login/Onboarding2FAInput.tsx:40]: let's make sure to useCallback

File: apps/mobile/src/screens/Login/Onboarding2FAInput.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+  const handleKeyPress = (e: NativeSyntheticEvent<TextInputKeyPressEventData>, index: number) => {
+    if (e.nativeEvent.key === 'Backspace' && index > 0 && !code[index]) {
+      inputs.current[index - 1]?.focus();
+    }
+  };
```
Comment (Robinnnnn) [inline apps/mobile/src/screens/Login/Onboarding2FAInput.tsx:46]: same here

File: apps/mobile/src/screens/Login/Onboarding2FAInput.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+      if (inputs.current[5]) {
+        inputs.current[5]?.focus();
+      }
+
+      onCodeComplete(newCode.join(''));
+    } else {
+      Alert.alert('Invalid Code', 'Clipboard content is not a 6-digit code.');
+    }
+  };
```
Comment (Robinnnnn) [inline apps/mobile/src/screens/Login/Onboarding2FAInput.tsx:62]: sameee

File: apps/mobile/src/screens/Login/Onboarding2FAInput.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+  const handlePaste = async () => {
+    const clipboardContent = await Clipboard.getString();
+    if (clipboardContent.length === 6) {
+      const newCode = clipboardContent.split('');
+      setCode(newCode);
+
+      if (inputs.current[5]) {
+        inputs.current[5]?.focus();
+      }
```
Comment (Robinnnnn) [inline apps/mobile/src/screens/Login/Onboarding2FAInput.tsx:56]: we can ditch the `if` and just go `inputs.current[5]?.focus();`

**Review by Robinnnnn — APPROVED** [review]

**Pull request merged**